src/platform_linux/platform_inline.h

```diff
@@ -34,11 +36,13 @@ platform_checksum_is_equal(checksum128 left, checksum128 right)
    return XXH128_isEqual(left, right);
 }
 
-static inline void
-platform_free_from_heap(platform_heap_id UNUSED_PARAM(heap_id), void *ptr)
-{
-   free(ptr);
-}
+static void
+platform_free_from_heap(platform_heap_id UNUSED_PARAM(heap_id),
```

Comment:
`heap_id` is really used in the new version, so this should go away.

Surprisingly -- no compiler complained about this inconsistency. (I'll let this be, and will fix this in part-3.)